.github/workflows/rust.yml

```diff
@@ -0,0 +1,156 @@
+
+defaults:
+  run:
+    working-directory: wren-modeling-rs
+
+jobs:
+  # Check crate compiles
+  linux-build-lib:
+    name: cargo check
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-builder
+        with:
+          rust-version: stable
+
+      - name: Cache Cargo
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ./wren-modeling-rs/target/
+          # this key equals the ones on `linux-build-lib` for re-use
+          key: cargo-cache-benchmark-${{ hashFiles('wren-modeling-rs/Cargo.toml') }}
+
+      - name: Check datafusion-common without default features
+        run: cargo check --all-targets
+
+  # Run tests
+  linux-test:
+    name: cargo test (amd64)
+    needs: [ linux-build-lib ]
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-builder
+        with:
+          rust-version: stable
+      - name: Run tests (excluding doctests)
+        run: cargo test --lib --tests --bins
+      - name: Verify Working Directory Clean
+        run: git diff --exit-code
+
+  windows:
+    name: cargo test (win64)
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-windows-builder
+      - name: Run tests (excluding doctests)
+        shell: bash
+        run: |
+          cargo test --lib --tests --bins
+
+  macos:
+    name: cargo test (macos)
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-macos-builder
+      - name: Run tests (excluding doctests)
+        shell: bash
+        run: cargo test --lib --tests --bins
+
+  macos-aarch64:
+    name: cargo test (macos-aarch64)
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-macos-aarch64-builder
+      - name: Run tests (excluding doctests)
+        shell: bash
+        run: cargo test --lib --tests --bins
+
+  check-fmt:
+    name: Check cargo fmt
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/rust/setup-builder
+        with:
+          rust-version: stable
+      - name: Run
+        run: cargo fmt --all -- --check
+
+  clippy:
```

Comment:
Let the lint checker before the build and the test.

Reply:
I think there are some cache issues here. I just followed the DataFusion workflow and found they have a commit that refactors the workflow by placing those checks after the tests.
https://github.com/apache/datafusion/commit/4924786b3a9cdbcbfeca146732e6b6e87f135566

I prefer to follow their way.